modules/datasets/src/track_alov.cpp

```diff
@@ -0,0 +1,384 @@
+					data.push_back(objects);
+				}
+			}
+
+			//Load annotations
+			//Loop for all sections of ALOV300++ (14 sections)
+			int currDatasetID = 0;
+			for (int i = 0; i < 14; i++)
+			{
+				//Loop for all videos in section
+				for (int k = 0; k < sectionSizes[i]; k++)
+				{
+					currDatasetID++;
+
+					//Open dataset's ground truth (annotation) file
+					string annoPath = fullAnnoPath(rootPath, i, k);
+					ifstream annoList(annoPath.c_str());
+					if (!annoList.is_open())
+					{
+						printf("Error: Can't open annotation file *.ANN!!!\n");
+						break;
+					}
+
+					//Ground Truth data
+					int n = 0;
+					double	x1 = 0, y1 = 0,
+						x2 = 0, y2 = 0,
+						x3 = 0, y3 = 0,
+						x4 = 0, y4 = 0;
+
+					do
+					{
+						//Make ALOV300++ Object
+						string tmp;
+
+						getline(annoList, tmp);
+						std::istringstream in(tmp);
+						in >> n >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;
+
+						Ptr<TRACK_alovObj> currObj = data[currDatasetID-1][n-1];
+
+						currObj->gtbb.clear();
+						currObj->gtbb.push_back(Point2d(x1, y1));
+						currObj->gtbb.push_back(Point2d(x2, y2));
+						currObj->gtbb.push_back(Point2d(x3, y3));
+						currObj->gtbb.push_back(Point2d(x4, y4));
+
+					} while (!annoList.eof());
+				}
+			}
+
+            return;
+        }
+
+		void TRACK_alovImpl::loadDatasetAnnotatedOnly(const string &rootPath)
+		{
+			vector <int> datasetsLengths;
+			int currDatasetID = 0;
+
+			printf("ALOV300++ Annotated Dataset Initialization...\n");
+
+			//Loop for all sections of ALOV300++ (14 sections)
+			for (int i = 0; i < 14; i++)
+			{
+				//Loop for all videos in section
+				for (int k = 0; k < sectionSizes[i]; k++)
+				{
+					vector <Ptr<TRACK_alovObj> > objects;
+					currDatasetID++;
+
+					//Open dataset's ground truth (annotation) file
+					string annoPath = fullAnnoPath(rootPath, i, k);
+					ifstream annoList(annoPath.c_str());
+					if (!annoList.is_open())
+					{
+						printf("Error: Can't open annotation file *.ANN!!!\n");
+						break;
+					}
+
+					int framesNum = 0;
+
+					do
+					{
+						//Make  ALOV300++ Object
+						Ptr<TRACK_alovObj> currObj(new TRACK_alovObj);
+						string tmp;
+						framesNum++;
+
+						//Ground Truth data
+						int	n = 0;
+						double x1 = 0, y1 = 0,
+							x2 = 0, y2 = 0,
+							x3 = 0, y3 = 0,
+							x4 = 0, y4 = 0;
+
+						getline(annoList, tmp);
+						std::istringstream in(tmp);
+						in >> n >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;
+
+						currObj->gtbb.push_back(Point2d(x1, y1));
+						currObj->gtbb.push_back(Point2d(x2, y2));
+						currObj->gtbb.push_back(Point2d(x3, y3));
+						currObj->gtbb.push_back(Point2d(x4, y4));
+
+						string fullPath = fullFramePath(rootPath, i, k, n);
+						if (!fileExists(fullPath))
+							break;
+
+						currObj->imagePath = fullPath;
+						currObj->id = n;
+
+						//Add object to storage
+						objects.push_back(currObj);
+
+					} while (!annoList.eof());
```

Comment:
It is better to use `annoList.good()` instead of `!annoList.eof()` since the former check also the status of `failbit` and `badbit`